helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -121,6 +122,9 @@
   private final RequestDispatcher dispatcher;
   private final AtomicBoolean versionWarningLogged = new AtomicBoolean();
 
+  private final TypeReference<Map<JobId, Job>> jobIdMap = new TypeReference<Map<JobId, Job>>() {
```

Comment:
good point; originally I did not have the three overloads of `jobs()` calling each other, so it was repeated three times in there before I condensed the logic down.